src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -168,6 +168,41 @@
                                        ? new CommitLogSegmentManagerCDC(c, DatabaseDescriptor.getCommitLogLocation())
                                        : new CommitLogSegmentManagerStandard(c, DatabaseDescriptor.getCommitLogLocation());
 
+    static void updateInitialized(boolean clientInitialized, boolean toolInitialized, boolean daemonInitialized)
+    {
+        DatabaseDescriptor.clientInitialized = clientInitialized;
+        DatabaseDescriptor.toolInitialized = toolInitialized;
+        DatabaseDescriptor.daemonInitialized = daemonInitialized;
+    }
+
+    /**
+     * This method can be called from tests if we need to re-initialize the DD
+     * again with a new config, partitioner or snitch. Note that this method will
+     * not delete directories or null the config, it simply resets the variables
+     * that prevent changing config or directories.
+     * <p/>
+     * It cannot set the config to null because this would cause NPEs in the
+     * schedulers that are still running, e.g. TPC.
```

Comment:
that's no longer relevant. perhaps the config can be set to null in C*?

Reply:
Hmm, yes the reference to TPC should be removed. Maybe it's not necessary to mention setting config at all since this reset code doesn't intend to change config - why mention what isn't a goal? 

If it were a goal, I'd think maybe setting it to `new Config()` would be considered to reset a default setup; `null` just seems like asking for trouble since there are many places that make the assumption it's been set.